WooCommerce/WooCommerceUITests/Flows/MockDataReader.swift

```diff
@@ -10,22 +10,43 @@ class GetMocks {
         "outofstock": "out of stock"
     ]
 
-    static func getProductsMockDataContent(withFilename filename: String) -> Data {
-        let json = Bundle(for: ProductsTests.self).url(forResource: filename, withExtension: "json")!
+    var productName = [
+        2123: "malaya shades",
+        2129: "akoya pearl shades",
+        2130: "black coral shades",
+        2131: "colorado shades",
+        2132: "rose gold shades"
+    ]
+
+    static func getMockData(test: AnyClass, filename file: String) -> Data {
+        let json = Bundle(for: test).url(forResource: file, withExtension: "json")!
 
         return try! Data(contentsOf: json)
     }
 
     static func readProductsData() throws -> [ProductData] {
-        let wrappedProductsData = try JSONDecoder().decode(MockFile.self, from: GetMocks.getProductsMockDataContent(withFilename: "products"))
-        var unwrappedProductsData = wrappedProductsData.response.jsonBody.data
+        let originalData = try JSONDecoder().decode(ProductMock.self, from: self.getMockData(test: ProductsTests.self, filename: "products"))
+        var updatedData = originalData.response.jsonBody.data
 
-        for index in 0..<unwrappedProductsData.count {
-            let rawStockStatus = unwrappedProductsData[index].stock_status
+        for index in 0..<updatedData.count {
+            let rawStockStatus = updatedData[index].stock_status
             let humanReadableStockStatus = GetMocks.init().stockStatus[rawStockStatus]!
-            unwrappedProductsData[index].stock_status = humanReadableStockStatus
+            updatedData[index].stock_status = humanReadableStockStatus
+        }
+
+        return updatedData
+    }
+
+    static func readReviewsData() throws -> [ReviewData] {
```

Comment:
This is pretty much exactly the same as above but for the reviews screen, not a common method, because it would end up being one with a long list of parameters (almost every line, is a different value) and the return type is different.

Reply:
👍 

If you fancy, I think this comment would be valuable in the code, too. It would prevent a developer eager to apply DRY to embark on an attempt that would reveal fruitless because of the many different values fact you pointed out